Probesys/bileto
# new: Allow to record time spent

## Related issue(s)


https://github.com/Probesys/bileto/issues/35

## Changes



- add permissions to see/create time spent
- create an entity TimeSpent (+ repo, migration, factory)
- add an input in the answer form to add time spent
- show the time spent in the ticket's timeline
- use the TimeSpent of a contract to calculate its consumed hours

## How to test manually



- create a ticket
- attach a contract to it
- answer to the ticket and add time (make sure it's **lower** than the max hours of the contract) → check it appears in the timeline
- answer to the ticket and add time (make sure it's **greater** than the remaining hours of the contract) → check that the time is split and part of it is not charged
- check that the contract is finished and that all the time is consumed

## Checklist
